examples/Using_tool_required_for_customer_service.ipynb

```diff
@@ -0,0 +1,515 @@
+    "        print(f\"Calling function {function_name}\")\n",
+    "        function_arguments = json.loads(function_call.function.arguments)\n",
+    "    \n",
+    "        if function_name == 'get_instructions':\n",
+    "\n",
+    "            respond = False\n",
+    "    \n",
+    "            instruction_name = function_arguments['problem']\n",
+    "            instructions = INSTRUCTIONS['type' == instruction_name]\n",
+    "    \n",
+    "            messages.append(\n",
+    "                                {\n",
+    "                                    \"tool_call_id\": function_id,\n",
+    "                                    \"role\": \"tool\",\n",
+    "                                    \"name\": function_name,\n",
+    "                                    \"content\": instructions['instructions'],\n",
+    "                                }\n",
+    "                            )\n",
+    "    \n",
+    "        elif function_name != 'get_instructions':\n",
+    "\n",
+    "            respond = True\n",
+    "    \n",
+    "            messages.append(\n",
+    "                                {\n",
+    "                                    \"tool_call_id\": function_id,\n",
+    "                                    \"role\": \"tool\",\n",
+    "                                    \"name\": function_name,\n",
+    "                                    \"content\": function_arguments['message'],\n",
+    "                                }\n",
+    "                            )\n",
+    "    \n",
+    "            print(f\"Assistant: {function_arguments['message']}\")\n",
+    "    \n",
+    "    return (respond, messages)\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ca6502e7-f664-43ba-b15c-962c69091633",
+   "metadata": {},
+   "source": [
+    "## Example\n",
+    "\n",
+    "To test this we will run an example for a customer who has experienced fraud, and see how the model handles it.\n",
+    "\n",
+    "Play the role of the user and provide plausible next steps to keep the conversation going."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 230,
+   "id": "bb1530e4-dd82-4560-bd60-9cc9ac0dab73",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "User: Hi, I have had an item stolen that was supposed to be delivered to me yesterday.\n",
+      "Calling function get_instructions\n",
+      "Calling function speak_to_user\n",
+      "Assistant: I'm sorry to hear about the stolen item. Could you please provide me with more details about the fraudulent activity, including the date and the items involved? This information will help us to investigate further and process any necessary actions, such as a refund.\n"
+     ]
+    }
+   ],
+   "source": [
+    "messages = submit_user_message(\"Hi, I have had an item stolen that was supposed to be delivered to me yesterday.\",messages)"
```

Comment:
messages is not defined

Reply:
Good catch, updated - not defining it in first cell so the function uses the default.